include/api/Metrics.h

```diff
@@ -289,10 +280,126 @@ class Metrics
 
   mutable std::mutex _mutex;
   LookupTable _lookups;
-  MetricBlobs _blobs;
+  BlobStorage _blobs;
   uint16_t _cur_blob = 0;
   uint16_t _cur_off  = 0;
 
+public:
+  // These are sort of factory classes, using the Metrics singleton for all storage etc.
+  class Gauge
+  {
+  public:
+    using self_type = Gauge;
+    using SpanType  = Metrics::SpanType;
+
+    class AtomicType : public Metrics::AtomicType
+    {
+    };
+
+    static Metrics::IdType
+    create(const std::string_view name)
+    {
+      auto &instance = Metrics::instance();
+
+      return instance._create(name);
+    }
+
+    static AtomicType *
+    createPtr(const std::string_view name)
```

Comment:
const std::string_view is overkill, all the member functions are const I think.

Reply:
does it hurt? Other than your eyes ?

Reply:
Alan will sentence you to hard labor on his pumpkin farm for this.